cdap-common/src/main/java/org/apache/twill/internal/yarn/Hadoop21YarnAppClient.java

```diff
@@ -126,22 +154,25 @@ private Resource adjustMemory(GetNewApplicationResponse response, Resource capab
 
   @Override
   public ProcessController<YarnApplicationReport> createProcessController(ApplicationId appId) {
-    return new ProcessControllerImpl(yarnClient, appId);
+    return new ProcessControllerImpl(getYarnClient(), appId);
   }
 
   @Override
   public List<NodeReport> getNodeReports() throws Exception {
-    return this.yarnClient.getNodeReports();
+    return getYarnClient().getNodeReports();
   }
 
   @Override
   protected void startUp() throws Exception {
-    yarnClient.start();
+    // no-op
```

Comment:
perhaps add a note here on why you don't need to do anything here. otherwise, move that comment to getYarnClient and remove this method?